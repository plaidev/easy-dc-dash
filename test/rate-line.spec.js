import Vue from 'vue';
import { mount, createLocalVue } from 'vue-test-utils';
import CssModuleTestHelperMixin from './helpers/css-modules-test-helper-mixin';

import Store from '../libs/store';
import RateLine from '@/rate-line.vue';

const localVue = createLocalVue();
localVue.use(CssModuleTestHelperMixin);

describe('rate-line', () => {
  Store.registerData([
    { d1: 'a', d2: true, v: 1 },
    { d1: 'a', d2: false, v: 2 },
    { d1: 'b', d2: false, v: 3 },
    { d1: 'b', d2: true, v: 5 }
  ]);

  const wrapper = mount(localVue.extend(RateLine), {
    localVue,
    propsData: {
      id: 'test-rate-line',
      dimension: 'd1',
      reduce: 'v'
    },
    attachToDocument: true
  });

  it('to match snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
