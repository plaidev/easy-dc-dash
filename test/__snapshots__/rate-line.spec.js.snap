// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rate-line to match snapshot 1`] = `
<div class="outer-container" style="width: 377px; height: 233px;">
    <div class="backdrop"></div>
    <div class="card-container">
        <div class="inner-container" style="width: 377px; height: 233px;">
            <div class="container-header">
                <div class="icon-box">
                    <i class="fa fa-window-maximize"></i>
                </div>
                <!---->
            </div>
            <div class="render-area">
                <div id="test-rate-line" class="krt-dc-component dc-chart" style="display: flex; position: relative; width: 100%; height: 100%;">
                    <!---->
                    <div class="reset-button">
                        <a class="reset" style="display: none;">
                            <i class="fa fa-ban">&nbsp;</i> <span>フィルター解除</span></a>
                    </div>
                    <!---->
                    <svg width="200" height="200">
                        <g>
                            <g class="grid-line horizontal" transform="translate(40,20)"></g>
                            <g class="grid-line vertical" transform="translate(40,20)">
                                <line x1="30" y1="150" x2="30" y2="0" opacity="0"></line>
                                <line x1="90" y1="150" x2="90" y2="0" opacity="0"></line>
                            </g>
                            <g class="chart-body" transform="translate(40, 20)" clip-path="url(about:blank#test-rate-line-clip)">
                                <g class="stack-list">
                                    <g class="stack _0">
                                        <path class="line" stroke="#1f77b4"></path>
                                        <text class="lineLabel" text-anchor="middle"></text>
                                        <text class="lineLabel" text-anchor="middle"></text>
                                    </g>
                                </g>
                                <g class="dc-tooltip-list">
                                    <g class="dc-tooltip _0">
                                        <path class="yRef" style="display: none;" stroke-dasharray="5,5"></path>
                                        <path class="xRef" style="display: none;" stroke-dasharray="5,5"></path>
                                        <circle class="dot" r="5" style="fill-opacity: 0.6; stroke-opacity: 0.6;" fill="#1f77b4"></circle>
                                        <circle class="dot" r="5" style="fill-opacity: 0.6; stroke-opacity: 0.6;" fill="#1f77b4"></circle>
                                    </g>
                                </g>
                            </g>
                            <g class="axis x" transform="translate(40,170)">
                                <g class="tick" style="opacity: 0.000001;" transform="translate(30,0)">
                                    <line y2="6"></line>
                                    <text dy=".71em" style="text-anchor: middle;" y="9">a</text>
                                </g>
                                <g class="tick" style="opacity: 0.000001;" transform="translate(90,0)">
                                    <line y2="6"></line>
                                    <text dy=".71em" style="text-anchor: middle;" y="9">b</text>
                                </g>
                                <path class="domain"></path>
                            </g>
                            <g class="axis y" transform="translate(40,20)">
                                <path class="domain"></path>
                            </g>
                        </g>
                        <defs>
                            <clippath id="test-rate-line-clip">
                                <rect width="120" height="150" transform="translate(-0, -0)"></rect>
                            </clippath>
                        </defs>
                    </svg>
                </div>
            </div>
        </div>
    </div>
</div>
`;
